pyproject.toml

```diff
@@ -36,7 +36,8 @@ python = "^3.7"
 osc-lib = "<=2.4.2"
 python-openstackclient = { version = "<=5.6.2", extras = ["cli"] }
 python-dateutil = "<=2.8.2"
-PyYAML = "~5.3"
+# Important to be on version 6+ to satisfy both cython and openstack upper-constraints 
+PyYAML = "6.0" 
```

Comment:
Xena upper-constraints will attempt to install [exactly the version from the upper-constraints file](https://github.com/openstack/requirements/blob/57e51fa16541a7468649d6658b6de5b4d835d0a4/upper-constraints.txt#L186).

However, outside of CI, the end user does not need to use upper-constraints at all, it is only used when installing multiple openstack services into the same environment for consistency (see devstack).

[Additionally, poetry's dependency specification recommends against exact constraints in pyproject.toml](https://python-poetry.org/docs/dependency-specification/).

To allow 5.4.1 in CI, but 6+ if unrestricted, we should specify something like: `PyYAML = ">= 5.4.1"`